test/system/252-quadlet.bats

```diff
@@ -1013,4 +1015,126 @@ EOF
     run_podman rmi $(pause_image)
 }
 
+@test "quadlet - image files" {
+    registry=localhost:${PODMAN_LOGIN_REGISTRY_PORT}
+    image_on_local_registry=$registry/quadlet_image_test:$(random_string)
+    authfile=$PODMAN_TMPDIR/authfile.json
+
+    # First, start the registry and populate the authfile that we can use for the test.
+    start_registry
+    run_podman login --authfile=$authfile \
+        --tls-verify=false \
+        --username ${PODMAN_LOGIN_USER} \
+        --password ${PODMAN_LOGIN_PASS} \
+        $registry
+
+    run_podman image tag $IMAGE $image_on_local_registry
+    run_podman image push --tls-verify=false --authfile=$authfile $image_on_local_registry
+
+    local image_for_test=$image_on_local_registry
+
+    # Remove the local image to make sure it will be pulled again
+    run_podman image rm --ignore $image_for_test
+
+    local quadlet_image_unit=image_test_$(random_string).image
+    local quadlet_image_file=$PODMAN_TMPDIR/$quadlet_image_unit
+    cat > $quadlet_image_file <<EOF
+[Image]
+Image=$image_for_test
+AuthFile=$authfile
+TLSVerify=false
+EOF
+
+    # Use the same directory for all quadlet files to make sure later steps access previous ones
+    local quadlet_tmpdir=$PODMAN_TMPDIR/quadlets
+    mkdir $quadlet_tmpdir
+
+    # Have quadlet create the systemd unit file for the image unit
+    run_quadlet "$quadlet_image_file" "$quadlet_tmpdir"
+    # Save the image service name since the variable will be overwritten
+    local image_service=$QUADLET_SERVICE_NAME
+
+    local quadlet_volume_unit=image_test_$(random_string).volume
+    local quadlet_volume_file=$PODMAN_TMPDIR/$quadlet_volume_unit
+    cat > $quadlet_volume_file <<EOF
+[Volume]
+Driver=image
+Image=$quadlet_image_unit
+EOF
+
+    # Have quadlet create the systemd unit file for the image unit
+    run_quadlet "$quadlet_volume_file" "$quadlet_tmpdir"
+    # Save the image service name since the variable will be overwritten
+    local volume_service=$QUADLET_SERVICE_NAME
+    local volume_name=systemd-$(basename $quadlet_volume_file .volume)
```

Comment:
Not worth a repush, but maybe this might be more readable if it was moved up 12 lines:
```bash
    local volume_name=image_test_$(random_string)
    local quadlet_volume_unit=${volume_name}.volume
    local quadlet_volume_file=...
```